volatility3/framework/plugins/windows/etwpatch.py

```diff
@@ -0,0 +1,128 @@
+        return [
+            (vad_start, vad_size, proc.add_process_layer())
+            for vad_start, vad_size, filepath in collected_modules
+            if pe_symbols.PESymbols.filename_for_path(filepath) == dll_name
+        ]
+
+    def _find_symbols(self, dll_name, symbols, proc_layer_name, dll_vads):
+        """Find symbols for a specific DLL."""
+        filter_module = {dll_name: {"names": symbols}}
+        process_modules = {dll_name: [(proc_layer_name, vad_start, vad_size) for vad_start, vad_size, _ in dll_vads]}
+        return pe_symbols.PESymbols.find_symbols(self.context, self.config_path, filter_module, process_modules)
+
+    def _get_first_opcode(self, proc_layer_name, function_start):
+        """Check the first opcode of a function."""
+        try:
+            return self.context.layers[proc_layer_name].read(function_start, 1).hex()
+        except exceptions.InvalidAddressException:
+            return None
+
+    def _generator(self):
+        pid_filter = self.config.get('pid', None)
+
+        for proc in pslist.PsList.list_processes(
+                context=self.context,
+                kernel_module_name=self.config['kernel']):
+            # Skip processes not in the PID filter
+            if pid_filter and proc.UniqueProcessId not in pid_filter:
+                continue
+
+            pid = int(proc.UniqueProcessId)
+            proc_name = proc.ImageFileName.cast(
+                "string",
+                max_length=proc.ImageFileName.vol.count,
+                errors='replace'
+            )
+
+            try:
+                proc_layer_name = proc.add_process_layer()
+            except exceptions.InvalidAddressException:
+                continue
+
+            dlls_to_check = {
+                "ntdll.dll": [
+                    "EtwEventWrite",
+                    "EtwEventWriteFull",
+                    "NtTraceEvent"
+                ],
+                "advapi32.dll": [
+                    "EventWrite"
+                ]
+            }
+
+            for dll_name, symbols in dlls_to_check.items():
+                dll_vads = self._get_dll_vads(proc, dll_name)
+                if not dll_vads:
+                    continue
+
+                found_symbols, _ = self._find_symbols(dll_name, symbols, proc_layer_name, dll_vads)
+                if dll_name not in found_symbols:
+                    continue
+
+                for symbol_name, function_start in found_symbols[dll_name]:
+                    opcode = self._get_first_opcode(proc_layer_name, function_start)
+                    if opcode in ('c3', 'e9'):  # RET or JMP
+                        yield (0, (
+                            pid,
+                            proc_name,
+                            dll_name,
+                            symbol_name,
+                            format_hints.Hex(function_start),
+                            opcode
```

Comment:
It would be better to yield the string name of the instruction (RET or JMP) instead of the hex opcode to aid people less familiar with reversing.